File: tests/java/com/trulioo/sdk/model/RecordTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;

public class RecordTest {
    private Record record;

    @BeforeEach
    public void beforeEach() {
        record = new Record();
    }

    @Test
    public void testTransactionRecordID() {
        String transactionRecordID = "test-transaction-guid";
        record.setTransactionRecordID(transactionRecordID);

        assertEquals(transactionRecordID, record.getTransactionRecordID());
        assertEquals(transactionRecordID,
                new Record().transactionRecordID(transactionRecordID).getTransactionRecordID());
    }

    @Test
    public void testRecordStatus() {
        String recordStatus = "match";
        record.setRecordStatus(recordStatus);

        assertEquals(recordStatus, record.getRecordStatus());
        assertEquals(recordStatus, new Record().recordStatus(recordStatus).getRecordStatus());
    }

    @Test
    public void testDatasourceResults() {
        record.addDatasourceResultsItem(new DatasourceResult());
        record.addDatasourceResultsItem(new DatasourceResult());
        List<DatasourceResult> datasourceResults = Arrays.asList(new DatasourceResult().datasourceStatus("match"));
        record.setDatasourceResults(datasourceResults);

        assertEquals(datasourceResults, record.getDatasourceResults());
        assertEquals(datasourceResults, new Record().datasourceResults(datasourceResults).getDatasourceResults());
    }

    @Test
    public void testErrors() {
        record.addErrorsItem(new ServiceError());
        record.addErrorsItem(new ServiceError());
        List<ServiceError> errors = Arrays.asList(new ServiceError().message("test"));
        record.setErrors(errors);

        assertEquals(errors, record.getErrors());
        assertEquals(errors, new Record().errors(errors).getErrors());
    }

    @Test
    public void testRule() {
        RecordRule rule = new RecordRule().ruleName("rule1");
        record.setRule(rule);

        assertEquals(rule, record.getRule());
        assertEquals(rule, new Record().rule(rule).getRule());
    }

    @Test
    public void testEquals() {
        String transactionRecordID = "test-transaction-guid";
        Record record1 = new Record().transactionRecordID(transactionRecordID);

        assertEquals(record1, record1);
        assertEquals(record1, new Record().transactionRecordID(transactionRecordID));
        assertNotEquals(record1, new Record().transactionRecordID(transactionRecordID + "1"));
        assertNotEquals(record1, null);
    }

    @Test
    public void testHashcode() {
        String transactionRecordID = "test-transaction-guid";

        assertEquals(new Record().transactionRecordID(transactionRecordID).hashCode(),
                new Record().transactionRecordID(transactionRecordID).hashCode());
    }

    @Test
    public void testToString() {
        String transactionRecordID = "test-transaction-guid";

        assertEquals(new Record().transactionRecordID(transactionRecordID).toString(),
                new Record().transactionRecordID(transactionRecordID).toString());
    }
}

File: tests/unit/php/Model/RecordTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use Trulioo\SDK\Model\DatasourceResult;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\Record;
use Trulioo\SDK\Model\RecordRule;
use Trulioo\SDK\Model\ServiceError;

class RecordTest extends TestCase
{
    private Record $record;

    public function setUp(): void
    {
        $this->record = new Record([]);
    }

    public function testPropertyTransactionRecordId()
    {
        $this->record->setTransactionRecordId('test-transaction-guid');
        $this->assertEquals('test-transaction-guid', $this->record->getTransactionRecordId());
    }

    public function testPropertyRecordStatus()
    {
        $this->record->setRecordStatus('match');
        $this->assertEquals('match', $this->record->getRecordStatus());
    }

    public function testPropertyDatasourceResults()
    {
        $this->record->setDatasourceResults([
            new DatasourceResult([
                'datasource_status' => 'match',
                'datasource_name' => 'source1'
            ])
        ]);
        $this->assertEquals(
            [
                new DatasourceResult([
                    'datasource_status' => 'match',
                    'datasource_name' => 'source1'
                ])
            ],
            $this->record->getDatasourceResults()
        );
    }

    public function testPropertyErrors()
    {
        $this->record->setErrors([
            new ServiceError([
                'message' => 'test'
            ])
        ]);
        $this->assertEquals(
            [
                new ServiceError([
                    'message' => 'test'
                ])
            ],
            $this->record->getErrors()
        );
    }

    public function testPropertyRule()
    {
        $this->record->setRule(new RecordRule([
            'rule_name' => 'rule1'
        ]));
        $this->assertEquals(
            new RecordRule([
                'rule_name' => 'rule1'
            ]),
            $this->record->getRule()
        );
    }
}
